compiler-cli/src/dependencies.rs

```diff
@@ -508,7 +508,11 @@ fn get_manifest<Telem: Telemetry>(
 
     // If the config has unchanged since the manifest was written then it is up
     // to date so we can return it unmodified.
-    if manifest.requirements == config.all_dependencies()? {
+    if is_same_requirements(
+        manifest.clone().requirements,
+        config.clone().all_dependencies()?,
```

Comment:
I've suggested that this method only borrows rather than taking ownership so these clones are no longer required, but for reference if items are to be cloned it should always be the smallest clone possible. Here the whole parent container is cloned and then the rest of it is immediately deallocated, which is wasted work.